packages/contracts-bedrock/src/dispute/PermissionedDisputeGame.sol

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { FaultDisputeGame, IFaultDisputeGame, IBigStepper, IInitializable } from "src/dispute/FaultDisputeGame.sol";
+import "src/libraries/DisputeTypes.sol";
+import "src/libraries/DisputeErrors.sol";
+
+/// @title PermissionedDisputeGame
+/// @notice PermissionedDisputeGame is a contract that inherits from `FaultDisputeGame`, and contains two roles:
+///         - The `challenger` role, which is allowed to challenge a dispute.
+///         - The `proposer` role, which is allowed to create proposals and participate in their game.
+contract PermissionedDisputeGame is FaultDisputeGame {
+    /// @notice The proposer role is allowed to create proposals and participate in the dispute game.
+    address internal immutable PROPOSER;
+    /// @notice The challenger role is allowed to participate in the dispute game.
+    address internal immutable CHALLENGER;
+
+    /// @notice Modifier that gates access to the `challenger` and `proposer` roles.
+    modifier onlyAuthorized() {
+        if (!(msg.sender == PROPOSER || msg.sender == CHALLENGER)) {
+            revert BadAuth();
+        }
+        _;
+    }
+
+    /// @param _gameType The type ID of the game.
+    /// @param _absolutePrestate The absolute prestate of the instruction trace.
+    /// @param _genesisBlockNumber The block number of the genesis block.
+    /// @param _genesisOutputRoot The output root of the genesis block.
+    /// @param _maxGameDepth The maximum depth of bisection.
+    /// @param _splitDepth The final depth of the output bisection portion of the game.
+    /// @param _gameDuration The duration of the game.
+    /// @param _vm An onchain VM that performs single instruction steps on a fault proof program
+    ///            trace.
+    constructor(
+        GameType _gameType,
+        Claim _absolutePrestate,
+        uint256 _genesisBlockNumber,
+        Hash _genesisOutputRoot,
+        uint256 _maxGameDepth,
+        uint256 _splitDepth,
+        Duration _gameDuration,
+        IBigStepper _vm,
+        address _proposer,
+        address _challenger
+    )
+        FaultDisputeGame(
+            _gameType,
+            _absolutePrestate,
+            _genesisBlockNumber,
+            _genesisOutputRoot,
+            _maxGameDepth,
+            _splitDepth,
+            _gameDuration,
+            _vm
+        )
+    {
+        PROPOSER = _proposer;
+        CHALLENGER = _challenger;
```

Comment:
We may want to make these authorized roles so that we can reuse the same contracts for both whitelisted OR entirely permissioned. If there's an extended outage on the permissionless game for some reason governance may prefer a whitelisted contract than an entirely permissioned one. Going to ask about this.

Reply:
Agree that we may not want a sole challenger. Can follow up on this IMO.